pkg/clients/compute.go

```diff
@@ -72,7 +86,18 @@ func NewComputeClient(providerClient *gophercloud.ProviderClient, providerClient
 	if err != nil {
 		return nil, fmt.Errorf("failed to create compute service client: %v", err)
 	}
-	compute.Microversion = NovaMinimumMicroversion
+
+	// Make an initial version negotiation.
+	// The caller can redo this later with more restrictions via RequireMicroversion.
+	supportedVersions := []*utils.Version{}
+	for k := range NovaSupportedVersions {
+		supportedVersions = append(supportedVersions, NovaSupportedVersions[k])
+	}
+	version, _, err := utils.ChooseVersion(providerClient, supportedVersions)
```

Comment:
I think it would have to be `compute` rather than `providerClient` here? Not sure what `providerClient` would test here. Keystone? Nothing?

Except that it seems this interface needs a provider client. It makes me think ChooseVersion may not be the right API for microversions, which are an artifact of the Nova API specifically.

Reply:
Interesting! I will try to investigate this.

Reply:
Thank you for pointing this out! This function was just for setting the Keystone microversion (even though the docs indicated something different).